include/lbann/layers/loss/mean_squared_error.hpp

```diff
@@ -103,6 +103,41 @@ class mean_squared_error_layer : public data_type_layer<TensorDataType> {
   data_layout get_data_layout() const override { return T_layout; }
   El::Device get_device_allocation() const override { return Dev; }
 
+#ifdef LBANN_HAS_ONNX
```

Comment:
This class is a prime example of why `get_onnx_op_type` should not be public (see my general comment, point 2). This class does not have an override for that function (and looking at the impl of `fill_onnx_node`, it seems there's not a single right answer for what to return in that case). So there's inconsistent behavior; a user calling `get_onnx_op_type()`, which should be valid for this onnx-ified class, will get an exception, but `fill_onnx_node` will be fine.